File: 이것이 취업을 위한 코딩 테스트다/정렬/카드_정렬하기/카드_정렬하기.cpp
#include <iostream>
#include <queue>

using namespace std;

int n, result;
priority_queue<int> pq;

int main(void) {
  cin >> n;

  for (int i = 0; i < n; i += 1) {
    int x;
    cin >> x;
    pq.push(-x);
  }

  while (pq.size() != 1) {
    int one = -pq.top();
    pq.pop();
    int two = -pq.top();
    pq.pop();
    int sum_value = one + two;
    result += sum_value;
    pq.push(-sum_value);
  }

  cout << result << '\n';
}

File: 이것이 취업을 위한 코딩 테스트다/정렬/카드_정렬하기/카드_정렬하기.py
# n = int(input())
# nums = []

# for _ in range(n):
#   nums.append(int(input()))

# nums.sort()

# first = nums[0]
# total = 0

# for i in range(1, n):
#   first += nums[i]
#   total += first

# print(total)

import heapq

n = int(input())

heap = []
for i in range(n):
  data = int(input())
  heapq.heappush(heap, data)

result = 0

while len(heap) != 1:
  one = heapq.heappop(heap)
  two = heapq.heappop(heap)
  sum_value = one + two
  result += sum_value
  heapq.heappush(heap, sum_value)

print(result)